sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -1145,11 +1145,22 @@ class MapboxMap internal constructor(
    * Performance benefit of the cache depends on the style as not all layers are cacheable due to e.g.
    * viewport aligned features. Render cache always prefers quality over performance.
    *
+   * If [RenderCacheOptions.size] is not specified explicitly or is less than zero when using [RenderCacheOptions.Builder] - render cache will be disabled.
+   *
    * @param options Options defining the render cache behavior
    */
   @MapboxExperimental
   fun setRenderCacheOptions(options: RenderCacheOptions) {
-    nativeMapWeakRef.call { this.renderCacheOptions = options }
+    options.size?.let { size ->
+      nativeMapWeakRef.call {
+        this.renderCacheOptions = if (size >= 0)
+          options
+        else
+          RenderCacheOptions.Builder().setDisabled().build()
```

Comment:
I think we should throw exceptions when the render cache is set to less than 0.

Reply:
In that case should we also throw an exception when `size` is not specified explicitly when building an object? IMHO it does not matter much until we have it all documented. Having `size == null || size < 0` ~ `cache disabled` seems pretty consistent to me.

Reply:
BTW I can still set cache to disabled if size < 0 but also log an error message @pengdev WDYT?

Reply:
IMO negative value is just wrong, user should never expect setting the size to a negative value should be permitted in any cases, I think we should throw illegal argument here.

Reply:
You convinced me 👍